lightning/src/ln/onchaintx.rs

```diff
@@ -638,27 +704,27 @@ impl<ChanSigner: ChannelKeys> OnchainTxHandler<ChanSigner> {
 
 		// Check if any pending claim request must be rescheduled
 		for (first_claim_txid, ref claim_data) in self.pending_claim_requests.iter() {
-			if claim_data.height_timer == height {
-				bump_candidates.insert(*first_claim_txid);
+			if let Some(h) = claim_data.height_timer {
+				if h == height {
+					bump_candidates.insert(*first_claim_txid, (*claim_data).clone());
+				}
 			}
 		}
 
 		// Build, bump and rebroadcast tx accordingly
 		log_trace!(self, "Bumping {} candidates", bump_candidates.len());
-		for first_claim_txid in bump_candidates.iter() {
-			if let Some((new_timer, new_feerate)) = {
-				if let Some(claim_material) = self.pending_claim_requests.get(first_claim_txid) {
-					if let Some((new_timer, new_feerate, bump_tx)) = self.generate_claim_tx(height, &claim_material, &*fee_estimator) {
-						log_trace!(self, "Broadcast onchain {}", log_tx!(bump_tx));
-						broadcaster.broadcast_transaction(&bump_tx);
-						Some((new_timer, new_feerate))
-					} else { None }
-				} else { unreachable!(); }
-			} {
-				if let Some(claim_material) = self.pending_claim_requests.get_mut(first_claim_txid) {
-					claim_material.height_timer = new_timer;
-					claim_material.feerate_previous = new_feerate;
-				} else { unreachable!(); }
+		let mut pending_claim_updates = Vec::with_capacity(bump_candidates.len());
+		for (first_claim_txid, claim_material) in bump_candidates.iter() {
+			if let Some((new_timer, new_feerate, bump_tx)) = self.generate_claim_tx(height, &claim_material, &*fee_estimator) {
+				log_trace!(self, "Broadcast onchain {}", log_tx!(bump_tx));
+				broadcaster.broadcast_transaction(&bump_tx);
+				pending_claim_updates.push((*first_claim_txid, new_timer, new_feerate));
+			}
+		}
+		for updates in pending_claim_updates {
```

Comment:
This seems to not be needed. At least if I revert most of this hunk at the end it still passes borrowck etc.

Reply:
Hmmmm? I hit a borrow checker with original hunk, you can't have both `pending_claim_requests` as mutable and `generate_claim_tx`

Reply:
Hmm, this diff compiles for me?

```
diff --git a/lightning/src/ln/onchaintx.rs b/lightning/src/ln/onchaintx.rs
index 2f08fe291..3b8238ddc 100644
--- a/lightning/src/ln/onchaintx.rs
+++ b/lightning/src/ln/onchaintx.rs
@@ -713,18 +713,14 @@ impl<ChanSigner: ChannelKeys> OnchainTxHandler<ChanSigner> {
 
                // Build, bump and rebroadcast tx accordingly
                log_trace!(self, "Bumping {} candidates", bump_candidates.len());
-               let mut pending_claim_updates = Vec::with_capacity(bump_candidates.len());
                for (first_claim_txid, claim_material) in bump_candidates.iter() {
                        if let Some((new_timer, new_feerate, bump_tx)) = self.generate_claim_tx(height, &claim_material, &*fee_estimator) {
                                log_trace!(self, "Broadcast onchain {}", log_tx!(bump_tx));
                                broadcaster.broadcast_transaction(&bump_tx);
-                               pending_claim_updates.push((*first_claim_txid, new_timer, new_feerate));
-                       }
-               }
-               for updates in pending_claim_updates {
-                       if let Some(claim_material) = self.pending_claim_requests.get_mut(&updates.0) {
-                               claim_material.height_timer = updates.1;
-                               claim_material.feerate_previous = updates.2;
+                               if let Some(claim_material) = self.pending_claim_requests.get_mut(first_claim_txid) {
+                                       claim_material.height_timer = new_timer;
+                                       claim_material.feerate_previous = new_feerate;
+                               }
                        }
                }
        }
```